File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.java
import java.io.*;
import java.util.*;

public class Main {
    static class Member implements Comparable<Member>{
        int age;
        String name;
        int idx;
        public Member(int age, String name,int idx) {
            this.age = age;
            this.name = name;
            this.idx = idx;
        }

        @Override
        public int compareTo(Member o) {
            if(age!=o.age){
                return age-o.age;
            }
            return idx - o.idx;
        }
    }
    
    public static void main(String[] args) throws IOException {
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt();
        Member[] members = new Member[N];
        for (int i = 0; i < N; i++) {
            members[i] = new Member(scan.nextInt(),scan.next(),i);

        }
        Arrays.sort(members);

        for (Member member : members) {
            System.out.println(member.age+" "+member.name);
        }
    }
}

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
n = int(input())
arr = []
for i in range(n):
    age,name = map(str,input().split())
    age = int(age)
    arr.append([age,name])
for i in range(n):
    arr[i].append(i)

arr.sort(key = lambda x:(x[0],x[2]))
for i in arr:
    print(i[0],i[1])